Structure_Engine/Query/ReinforcementLayout.cs

```diff
@@ -178,6 +288,60 @@ public static List<Line> ReinforcementLayout(this LongitudinalReinforcement rein
             return rebarLines;
         }
 
+        /***************************************************/
+
+        [Description("Gets the TransverseReinforcement centrelines as a list of polylines, based on inner and outer profile edges and Bar parameters.")]
+        [Input("reinforcement", "The TransverseReinforcement to extract the centrelines from.")]
+        [Input("cover", "Rebar cover.", typeof(Length))]
+        [Input("outerProfileEdges", "The outer profile edges of the ConcreteSection to be populated.")]
+        [Input("innerProfileEdges", "The inner profile edges, or openings, of the ConcreteSection to be populated.")]
+        [Input("length", "Length of the host Bar used to generate the lines", typeof(Length))]
+        [Input("transformation", "Transformation needed to move the lines from the position of the host element.")]
+        [Output("points", "The centrelines of the LongitudinalReinforcement.")]
+        public static List<ICurve> ReinforcementLayout(this TransverseReinforcement reinforcement, double cover, List<ICurve> outerProfileEdges, List<ICurve> innerProfileEdges, double length, TransformMatrix transformation)
+        {
+            List<ICurve> rebarLines = new List<ICurve>();
+            List<ICurve> stirrupOutline = reinforcement.CenterlineLayout.ICurveLayout(outerProfileEdges, innerProfileEdges);
+
+            Vector dir = Vector.ZAxis.Transform(transformation);
+            double stirrupRange = (reinforcement.EndLocation - reinforcement.StartLocation) * length - (2 * cover + reinforcement.Diameter);
+            int count = (int)Math.Floor(stirrupRange / reinforcement.Spacing);
+
+            double spacing = reinforcement.Spacing;
+            if (reinforcement.AdjustSpacingToFit)
+                spacing = stirrupRange / count;
```

Comment:
Think we might want to use `Math.Ceiling` instead of floor for the case where we adjust the spacing to fit. Then the spacing can be seen as maximum spacing allowed, rather than minimum, for the case where automatic adjustment is allowed. This should then be reflected in the description of the Property on that object.